File: sample/src/main/java/ru/terrakok/cicerone/sample/mvp/bottom/forward/ForwardPresenter.java
package ru.terrakok.cicerone.sample.mvp.bottom.forward;

import com.arellomobile.mvp.InjectViewState;
import com.arellomobile.mvp.MvpPresenter;
import com.github.terrakok.cicerone.Router;

import ru.terrakok.cicerone.sample.Screens;

/**
 * Created by terrakok 26.11.16
 */

@InjectViewState
public class ForwardPresenter extends MvpPresenter<ForwardView> {
    private String container;
    private Router router;
    private int number;

    public ForwardPresenter(String container, Router router, int number) {
        this.container = container;
        this.router = router;
        this.number = number;

        getViewState().setChainText(createChain(number));
    }

    private String createChain(int number) {
        String chain = "[0]";

        for (int i = 0; i < number; i++) {
            chain += "➔" + (i + 1);
        }

        return chain;
    }

    public void onForwardPressed() {
        router.navigateTo(Screens.forwardScreen(container, number + 1), true);
    }

    public void onGithubPressed() {
        router.navigateTo(Screens.githubScreen(), true);
    }

    public void onBackPressed() {
        router.exit();
    }
}

File: sample/src/main/java/ru/terrakok/cicerone/sample/mvp/bottom/forward/ForwardPresenter.kt
package ru.terrakok.cicerone.sample.mvp.bottom.forward

import com.arellomobile.mvp.InjectViewState
import com.arellomobile.mvp.MvpPresenter
import com.github.terrakok.cicerone.Router
import ru.terrakok.cicerone.sample.Screens.forwardScreen
import ru.terrakok.cicerone.sample.Screens.githubScreen

/**
 * Created by terrakok 26.11.16
 */
@InjectViewState
class ForwardPresenter(
        private val container: String,
        private val router: Router,
        private val number: Int
) : MvpPresenter<ForwardView>() {

    private fun createChain(number: Int): String {
        var chain = "[0]"
        for (i in 0 until number) {
            chain += "➔" + (i + 1)
        }
        return chain
    }

    fun onForwardPressed() {
        router.navigateTo(forwardScreen(container, number + 1), true)
    }

    fun onGithubPressed() {
        router.navigateTo(githubScreen(), true)
    }

    fun onBackPressed() {
        router.exit()
    }

    init {
        viewState?.setChainText(createChain(number))
    }
}